src/Microsoft.DocAsCode.Build.SchemaDriven/SchemaDrivenDocumentProcessor.cs

```diff
@@ -31,6 +31,8 @@ public class SchemaDrivenDocumentProcessor
         private readonly bool _allowOverwrite;
         private readonly MarkdigMarkdownService _markdigMarkdownService;
         private readonly FolderRedirectionManager _folderRedirectionManager;
+        private readonly string _liveSiteHostName;
+        private readonly HashSet<string> _schemaToProcessRemoveHostName = new HashSet<string> { "Achivement", "ModuleAvailability", "ContextObject", "ContentNav" };
```

Comment:
To avoid maintaining this list, we can apply this logic to all SDP schemas. I feel it is acceptable to duplicate to logic here, although the processer will update the links again in post processor for SDP targeting HTML.